### PR TITLE
Adds the ability to choose the folder of cloned VM

### PR DESCRIPTION
**Pull request by galindro**
Adds the ability to choose the folder of cloned VM (ansible/ansible#10326):

``` yaml
- vsphere_guest:
    vcenter_hostname: "vcenter"
    username: "root"
    password: "pass"
    guest: "test"
    from_template: "yes"
    template_src: "test_template"
    cluster: "cluster_1"
    resource_pool: "/Resources"
    vm_extra_config:
      folder: "Test_Folder"
```

### CODE REVIEW

**Comment by gregdek** [comment]
@galindro looks like this fails merge testing, can you rebase?

**Comment by galindro** [comment]
Done @gregdek

File: cloud/vmware/vsphere_guest.py [hunk 1]
```diff
@@ -153,18 +153,11 @@
...
-        network_type: dvs
     vm_hardware:
       memory_mb: 2048
       num_cpus: 2
       osid: centos64Guest
       scsi: paravirtual
-      vm_cdrom:
-        type: "iso"
-        iso_path: "DatastoreName/cd-image.iso"
     esxi:
```
Comment (bcoca) [inline cloud/vmware/vsphere_guest.py:168]: why delete these parts of the examples?

**Comment by galindro** [comment]
I've not deleted. Maybe the rebase isn't ok as I expected. I'll try to rebase again.

**Comment by galindro** [comment]
How could I fix this? I can't make the rebase again. I could make a new commit with these lines, but I don't know if this is the correct way to fix this inconsistency.

**Comment by bcoca** [comment]
it does not look like a git issue, there is only a single commit and it has the deletion.

**Comment by galindro** [comment]
@bcoca I know this. I'm not blaming git. This certainly was my fault. Now how could I fix this? Doing a new commit?

**Comment by bcoca** [comment]
seems like that, as current commit does undesired things

**Comment by galindro** [comment]
Sorry, I can't understand. Are you saying that I can make a new commit to fix that exclusion?

**Comment by bcoca** [comment]
make a new commit with only the changes you want to make.

**Comment by galindro** [comment]
@bcoca I'll make a commit adding those excluded lines, because the proposed fix already in the first commit. Agree?

**Comment by galindro** [comment]
Done. If this was not sufficient I will close this PR and make the last one.

**Comment by bcoca** [comment]
merge the commits or just create a new clean one

**Comment by gregdek** [comment]
This is still not merging cleanly. Let's close this PR and make a new one if it still applies. Thanks @galindro for the PR.

**Pull request closed without merging**